pkg/controller/node/controller_utils.go

```diff
@@ -40,45 +40,37 @@ const (
 	LargeClusterThreshold = 20
 )
 
-// deletePods will delete all pods from master running on given node, and return true
-// if any pods were deleted, or were found pending deletion.
-func deletePods(kubeClient clientset.Interface, recorder record.EventRecorder, nodeName, nodeUID string, daemonStore cache.StoreToDaemonSetLister) (bool, error) {
-	remaining := false
+func getPodsOfNode(kubeClient clientset.Interface, nodeName string) (*api.PodList, error) {
 	selector := fields.OneTermEqualSelector(api.PodHostField, nodeName)
 	options := api.ListOptions{FieldSelector: selector}
 	pods, err := kubeClient.Core().Pods(api.NamespaceAll).List(options)
 	if err != nil {
-		return remaining, err
+		return nil, err
 	}
+	return pods, nil
+}
 
-	if len(pods.Items) > 0 {
-		recordNodeEvent(recorder, nodeName, nodeUID, api.EventTypeNormal, "DeletingAllPods", fmt.Sprintf("Deleting all Pods from Node %v.", nodeName))
-	}
+// deletePod will delete pod that is running on given node from master,
+// and return true if pod was deleted, or was found pending deletion.
+func deletePod(kubeClient clientset.Interface, recorder record.EventRecorder, pod *api.Pod, nodeUID string, daemonStore cache.StoreToDaemonSetLister) (bool, error) {
+	recordNodeEvent(recorder, pod.Spec.NodeName, nodeUID, api.EventTypeNormal, "EvictingPod", fmt.Sprintf("Evicting Pod %s from Node %s.", pod.Name, pod.Spec.NodeName))
 
-	for _, pod := range pods.Items {
-		// Defensive check, also needed for tests.
-		if pod.Spec.NodeName != nodeName {
-			continue
-		}
-		// if the pod has already been marked for deletion, we still return true that there are remaining pods.
-		if pod.DeletionGracePeriodSeconds != nil {
-			remaining = true
-			continue
-		}
-		// if the pod is managed by a daemonset, ignore it
-		_, err := daemonStore.GetPodDaemonSets(&pod)
-		if err == nil { // No error means at least one daemonset was found
-			continue
-		}
+	// if the pod has already been marked for deletion, we still return true that there are remaining pods.
+	if pod.DeletionGracePeriodSeconds != nil {
+		return true, nil
+	}
+	// if the pod is managed by a daemonset, ignore it
+	_, err := daemonStore.GetPodDaemonSets(pod)
```

Comment:
I'd prefer to leave the code here for now. It is copied from existing deletePods() code, so safer to keep it for now. Maybe add a TODO to consider getting rid of it.